icinga2/synology-services.conf

```diff
@@ -1,15 +1,104 @@
-// `check_synology` example service configuration snippet for Icinga 2
+// `check_synology` full service configuration snippet for Icinga 2
 
-apply Service "syno-load" {
-  import "generic-service"
+// Baseline template. All service definitions will inherit this.
+template Service "synology-base" {
 
   check_command = "check_synology"
 
+  if (host.vars.synology_host) {
+    vars.synology_host = host.vars.synology_host
+  } else {
+    vars.synology_host = "$address$"
+  }
+
+  if (host.vars.synology_location) {
+    display_name += " at " + host.vars.synology_location
+  }
+
+  // Enable this to make the command run on the monitored host
+  // when different from the Icinga instance.
+  // command_endpoint   = host_name
+
+}
+
+apply Service "Synology system load" {
+  import "generic-service"
+  import "synology-base"
+
+  // Warn if system load is higher than X.
   vars.synology_mode = "load"
-  vars.synology_host = "$address$"
+  vars.synology_warning = host.vars.synology_load1_warning || 5
+  vars.synology_critical = host.vars.synology_load1_critical || 10
+
+  assign where host.vars.check_synology == true || host.vars.os == "DSM"
+}
+
+apply Service "Synology system memory" {
+  import "generic-service"
+  import "synology-base"
+
+  // Warn if available system memory is lower than %.
+  vars.synology_mode = "memory"
+  vars.synology_warning = host.vars.synology_memory_warning || 20
+  vars.synology_critical = host.vars.synology_memory_critical || 10
+
+  assign where host.vars.check_synology == true || host.vars.os == "DSM"
+}
+
+apply Service "Synology disk temperature" {
+  import "generic-service"
+  import "synology-base"
+
+  // Warn if disk drive temperature is higher than °C.
+  // https://www.hdsentinel.com/forum/viewtopic.php?f=32&t=1519
+  vars.synology_mode = "disk"
+  vars.synology_warning = host.vars.synology_hddtemp_warning || 43
+  vars.synology_critical = host.vars.synology_hddtemp_critical || 50
```

Comment:
Hi again,

I spent some minutes on further research on this topic and wanted to share the outcome with you.

The article [What’s a Good Operating Temperature For My Synology?](https://mariushosting.com/whats-a-good-operating-temperature-for-my-synology/) shares some numbers about _Manufacturer Normal Internal Operating Ranges_, which apparently are 0° – 65°C for HDD disks and 0°C to 70°C for SSD disks.

With kind regards,
Andreas.

Reply:
Following my reasoning in the comment above [^1], I've kept the 43/50°C numbers here instead of increasing them to the official ranges supplied by the manufacturers.

For handing that information over to the user, I've added it to the inline documentation.

```c++
  /*
  CRITICAL when disk status is `SystemPartitionFailed` or `Crashed`.
  WARNING/CRITICAL when disk drive temperature is higher than °C.

  Official *Manufacturer Normal Internal Operating Ranges* are
  0° – 65°C for HDD disks and 0°C to 70°C for SSD disks.

  - https://www.hdsentinel.com/forum/viewtopic.php?f=32&t=1519
  - https://mariushosting.com/whats-a-good-operating-temperature-for-my-synology/
  */
```

[^1]: https://github.com/wernerfred/check_synology/pull/21#discussion_r825451178